### ISSUE TITLE
Rule proposal: `no-exports-in-scripts`

### ISSUE THREAD

**Issue by zanminkian** [sindresorhus/eslint-plugin-unicorn#2513]
### Description

Currently, a js file starts with hashbang `#!` is script file. In script file, using `export`s is useless and confused.

### Fail

```js
#!/usr/bin/env node

export const foo = {};
```

### Pass

```js
export {}
```

```js
#!/usr/bin/env node

console.log('foo');
```

### Proposed rule name

no-exports-in-scripts

**Comment by fregante** [comment]
Somewhat related:

- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1661

This rule:

- report exports if it has side effects (detected via hashbang)

That rule:

- report side effects if it has exports

However this rule as defined is much easier to implement: 

1. find export statement
2. report if line 1 is hashbang

**Comment by sindresorhus** [comment]
Accepted

**Comment by zanminkian** [comment]
Changed my mind since Node supports `import.meta.main`. Allowing exports in scripts could be useful for testing.

```js
#!/usr/bin/env node
export function main() {
  // some codes need to be tested
}
if(import.meta.main) {
  main()
}
```
